agent/task_context.go

```diff
@@ -154,7 +173,11 @@ func (a *Agent) makeTaskConfig(ctx context.Context, tc *taskContext) (*internal.
 	}
 
 	var confPatch *patch.Patch
-	if evergreen.IsGitHubPatchRequester(tc.taskModel.Requester) {
+	requester := ""
+	if tc.taskConfig != nil && tc.taskConfig.Task != nil {
+		requester = tc.taskConfig.Task.Requester
```

Comment:
This line will never run in production because `tc.taskConfig` isn't populated until [`makeTaskConfig` returns and sets `tc.taskConfig`](https://github.com/evergreen-ci/evergreen/blob/a68cebae20ca387a5ba4d97cc086b044fc7cc6b1/agent/agent.go#L440-L445). It seems like you want to check `task` for the requester instead?